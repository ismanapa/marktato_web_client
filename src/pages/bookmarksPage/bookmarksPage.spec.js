import React from 'react';
import { shallow } from 'enzyme';
import CircularProgress from '@material-ui/core/CircularProgress';

import BookmarkList from './components/bookmarktList';
import BookmarksPage from './bookmarksPage';

jest.mock('./components/bookmarktList', () => 'BookmarktList');

describe('BookmarkPage', () => {
  it('Should match snapshot', () => {
    const props = {
      data: { bookmarks: [{ name: 'bookmark1' }, { name: 'bookmark2' }] },
      loading: false,
      error: null,
    };

    const wrapper = shallow(<BookmarksPage {...props}/>);

    expect(wrapper).toMatchSnapshot();
  });

  it('Should have loader when loading is true', () => {
    const props = {
      data: {},
      loading: true,
      error: null,
    };

    const wrapper = shallow(<BookmarksPage {...props}/>);
    const loader = wrapper.find(CircularProgress);

    expect(loader).toHaveLength(1);
  });

  it('Should print error when loading is done and error exists', () => {
    const props = {
      data: {},
      loading: false,
      error: new Error('network error'),
    };

    const wrapper = shallow(<BookmarksPage {...props}/>);

    expect(wrapper.contains(props.error.toString())).toBeTruthy();
  });

  it('Should print bookmark list when loading is done', () => {
    const props = {
      data: { bookmarks: [{ name: 'bookmark1' }, { name: 'bookmark2' }] },
      loading: false,
      error: null,
    };

    const wrapper = shallow(<BookmarksPage {...props}/>);
    const list = wrapper.find(BookmarkList).first();

    expect(list.props().bookmarks).toEqual(props.data.bookmarks);
  });
});
