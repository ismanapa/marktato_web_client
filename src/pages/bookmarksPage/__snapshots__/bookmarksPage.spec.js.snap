// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BookmarkPage Should match snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <BookmarksPage
    data={
      Object {
        "bookmarks": Array [
          Object {
            "name": "bookmark1",
          },
          Object {
            "name": "bookmark2",
          },
        ],
      }
    }
    error={null}
    loading={false}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {
      "children": <BookmarktList
        bookmarks={
          Array [
            Object {
              "name": "bookmark1",
            },
            Object {
              "name": "bookmark2",
            },
          ]
        }
      />,
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "bookmarks": Array [
          Object {
            "name": "bookmark1",
          },
          Object {
            "name": "bookmark2",
          },
        ],
      },
      "ref": null,
      "rendered": null,
      "type": "BookmarktList",
    },
    "type": Symbol(react.fragment),
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "children": <BookmarktList
          bookmarks={
            Array [
              Object {
                "name": "bookmark1",
              },
              Object {
                "name": "bookmark2",
              },
            ]
          }
        />,
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "bookmarks": Array [
            Object {
              "name": "bookmark1",
            },
            Object {
              "name": "bookmark2",
            },
          ],
        },
        "ref": null,
        "rendered": null,
        "type": "BookmarktList",
      },
      "type": Symbol(react.fragment),
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
